lib/hanami/container.rb

```diff
@@ -4,17 +4,23 @@
 
 module Hanami
   class Container
-    class Router < ::Hanami::Router
+    class Mounting
+      def self.load!(&blk)
+        new.instance_exec(&blk)
+      end
+
       def mount(app, options)
-        app = app.new(path_prefix: options.fetch(:at)) if hanami_app?(app)
-        super(app, options)
+        app.configuration.path_prefix options.fetch(:at) if hanami_app?(app)
       end
 
       private
 
       def hanami_app?(app)
         app.ancestors.include? Hanami::Application
       end
+
+      def method_missing(m, *args)
```

Comment:
@jodosha I wasn't sure what directives are allowed inside `Hanami::Container.configure`. If `mount` is the only directive we allow, this is not needed.

If there are other directives, I think I can also stub them using actual methods instead of `method_missing`.